database/skylink.go

```diff
@@ -72,34 +82,83 @@ func (db *DB) SkylinkFetch(ctx context.Context, sl string) (Skylink, error) {
 		return Skylink{}, sr.Err()
 	}
 	s := Skylink{}
-	err := sr.Decode(&s)
+	err = sr.Decode(&s)
 	if err != nil {
 		return Skylink{}, err
 	}
 	return s, nil
 }
 
+// SkylinkMarkPinned marks a skylink as pinned (or no longer unpinned), meaning
+// that Pinner should make sure it's pinned by the minimum number of servers.
+func (db *DB) SkylinkMarkPinned(ctx context.Context, skylink string) error {
+	sl, err := CanonicalSkylink(skylink)
+	if err != nil {
+		return ErrInvalidSkylink
+	}
+	filter := bson.M{"skylink": sl}
+	update := bson.M{"$set": bson.M{"unpin": false}}
+	opts := options.UpdateOptions{}
+	opts.SetUpsert(true)
+	_, err = db.staticDB.Collection(collSkylinks).UpdateOne(ctx, filter, update)
+	return err
+}
+
+// SkylinkMarkUnpinned marks a skylink as unpinned, meaning that all servers
+// should stop pinning it.
+func (db *DB) SkylinkMarkUnpinned(ctx context.Context, skylink string) error {
+	sl, err := CanonicalSkylink(skylink)
+	if err != nil {
+		return ErrInvalidSkylink
+	}
+	filter := bson.M{"skylink": sl}
+	update := bson.M{"$set": bson.M{"unpin": true}}
+	opts := options.UpdateOptions{}
+	opts.SetUpsert(true)
+	_, err = db.staticDB.Collection(collSkylinks).UpdateOne(ctx, filter, update)
+	return err
+}
+
 // SkylinkServerAdd adds a new server to the list of servers known to be pinning
 // this skylink. If the skylink does not already exist in the database it will
 // be inserted. This operation is idempotent.
-func (db *DB) SkylinkServerAdd(ctx context.Context, sl string, server string) error {
+func (db *DB) SkylinkServerAdd(ctx context.Context, skylink string, server string) error {
+	sl, err := CanonicalSkylink(skylink)
+	if err != nil {
+		return ErrInvalidSkylink
+	}
 	filter := bson.M{"skylink": sl}
 	update := bson.M{"$addToSet": bson.M{"servers": server}}
 	opts := options.UpdateOptions{}
 	opts.SetUpsert(true)
-	_, err := db.staticDB.Collection(collSkylinks).UpdateOne(ctx, filter, update, &opts)
+	_, err = db.staticDB.Collection(collSkylinks).UpdateOne(ctx, filter, update, &opts)
 	return err
 }
 
 // SkylinkServerRemove removes a server to the list of servers known to be
 // pinning this skylink. If the skylink does not exist in the database it will
 // not be inserted.
-func (db *DB) SkylinkServerRemove(ctx context.Context, sl string, server string) error {
+func (db *DB) SkylinkServerRemove(ctx context.Context, skylink string, server string) error {
```

Comment:
you could have the database simply handle `skymodules.Skylink` and take care of the validation higher up, you can resolve this comment if you feel that's not a good idea was just suggesting it because it kind of simplifies things I think (maybe) to do this in the handler itself.

E.g. 
```go
sl, err := validateSkylink(params.Skylink)
if err != nil { return 400 } 
err = db.skylinkRemove(sl)
if err != nil { return 500 } 
```
rather than
```go
err := db.skylinkRemove(params.Skylink)
if err != nil {
    if errors.Contains(err, ErrInvalidSkylink) {
        return 400
    }
    return 500
}
```

Reply:
That sounds good. It seems like you don't necessarily need it in this PR, so I'd rather add it to #7 in order to keep the conflicts down.